src/chain.cpp

```diff
@@ -5,6 +5,27 @@
 
 #include <chain.h>
 
+// Dogecoin: Include the single function signature we need, to avoid the entire
+// validation.h being pulled in and creating a cyclic include loop
+bool ReadBlockHeaderFromDisk(CBlockHeader& block, const CBlockIndex* pindex, const Consensus::Params& consensusParams, const bool fCheckPOW = true);
```

Comment:
Yeah, can split it out. Done now.

Reply:
Hmm it's still here?

Reply:
Misread and thought it was ReadBlockOrHeader I was cutting. I think there was an issue switching this, but let me give it a second go so at least I've got the rationale.

Reply:
My issue with this is that this is declared into chain.cpp with a note that if we don't declare this here, we'd need validation.h inclusion, but the function is not called anywhere? So if there is a circular inclusion, it can probably be solved by implementing things differently. Yet, since that code seems to not yet be introduced, if we merge this it will be hiding a problem rather than fixing a problem properly - but I have no idea what the problem is because there is no code to expose it.

Reply:
Actually properly removed it this time.